account_analytic_security/models/account_analytic_account.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# © 2017 Esther Martín - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    @api.multi
+    @api.depends('project_ids', 'project_ids.members',
+                 'project_ids.message_follower_ids')
+    def _compute_partner_ids(self):
+        for account in self:
+            for project in account.project_ids:
+                account.partner_ids = [(4, member.partner_id.id)
+                                       for member in project.members]
```

Comment:
yo quitaría todo lo que hay después del for, y haría lo siguiente:
partners = self.env['res.parter']
partners += project.mapped('menbers.ids')
partners += project.mapped('message_follower_ids.ids')
account.partner_ids = [(6, 0, partners.ids)]